code/drasil-printers/lib/Language/Drasil/Markdown/Print.hs

```diff
@@ -0,0 +1,344 @@
+lnl :: Doc
+lnl = text "\\\\\\\\"
+
+-----------------------------------------------------------------
+-------------------- TABLE PRINTING -----------------------------
+-----------------------------------------------------------------
+
+-- | Renders Markdown table, called by 'printLO'
+makeTable :: RefMap -> [[Spec]] -> Doc -> Bool -> Doc -> Doc
+makeTable _  [] _ _ _  = error "No table to print"
+makeTable rm ls r b t  = 
+  divTag r $$
+  makeHeaderCols (head matrix) sizes $$
+  makeRows (tail matrix) sizes <> nl $$
+  capt
+    where
+      matrix = mkDocMatrix rm ls
+      sizes = columnSize matrix
+      capt = if b then bold (caption t) <> nl else empty
+
+-- | Helper for creating a Doc matrix
+mkDocMatrix :: RefMap -> [[Spec]] -> [[Doc]]
+mkDocMatrix rm = map $ map (pSpec rm)
+
+-- | Helper for getting table column size
+columnSize :: [[Doc]] -> [Int]
+columnSize = map (maximum . map docLength) . transpose
+
+-- | Helper for creating table rows
+makeRows :: [[Doc]] -> [Int] -> Doc
+makeRows lls sizes = foldr (($$) . (`makeColumns` sizes)) empty lls
+
+-- | makeHeaderCols: Helper for creating table header row
+-- | makeColumns: Helper for creating table columns
+makeHeaderCols, makeColumns :: [Doc] -> [Int] -> Doc
+makeHeaderCols l sizes = header $$ seperators
+  where header     = pipe <> hcat (punctuate pipe (zipWith makeCell l sizes)) <> pipe        
+        seperators = pipe <> hcat (punctuate pipe (map makeDashCell sizes))   <> pipe
+
+makeColumns ls sizes = pipe <> hcat (punctuate pipe (zipWith makeCell ls sizes)) <> pipe
+
+-- | Helper for making table seperation row
+makeDashCell :: Int -> Doc
+makeDashCell size = text ":" <> text (replicate (size - 1) '-')
+
+-- | Helper for rendering a table cell
+makeCell :: Doc -> Int -> Doc
+makeCell content size = content <> spaces
+  where
+    numOfSpaces = size - docLength content
+    spaces      = text $ replicate numOfSpaces ' '
+
+-----------------------------------------------------------------
+-------------------- DEFINITION PRINTING ------------------------
+-----------------------------------------------------------------
+
+-- | Renders definition tables (Data, General, Theory, etc.)
+makeDefn :: RefMap -> [(String,[LayoutObj])] -> Doc -> Doc
+makeDefn _  [] _ = error "L.Empty definition"
+makeDefn rm ps l = 
+  makeDHeaderText rm ps l $$ 
+  makeHeaderCols [text "Refname", l] size $$ 
+  makeRows docDefn size
+  where
+    docDefn = mkDocDefn rm ps
+    size = columnSize docDefn
+
+-- | Helper for convering definition to Doc matrix
+mkDocDefn :: RefMap -> [(String,[LayoutObj])] -> [[Doc]]
+mkDocDefn rm = map (\(f, d) -> [text f, makeLO rm (f,d)])
+
+-- | Renders the title/header of the definition table
+makeDHeaderText :: RefMap -> [(String, [LayoutObj])] -> Doc -> Doc
+makeDHeaderText rm ps l = defnHTag header <> nl
+  where
+    lo = lookup "Label" ps
+    c = maybe l (\lo' -> makeLO rm ("Label", lo')) lo 
+    header = nl <> text "##" <+> heading c l <> nl
```

Comment:
Instead of using `text "##"`, I think it wold be good to use `h2` or something of the sorts.

Reply:
Addressed in https://github.com/JacquesCarette/Drasil/pull/3823/commits/88a109896dbccd86b0a148837adea09cb8b556a7. I created a `h'` function which fixes this issue and also creates a useful function that can be used when fixing the `Markdown` format rendering (also mentioned in https://github.com/JacquesCarette/Drasil/pull/3823#discussion_r1657652245)